src/test/java/seedu/address/model/leave/LeaveTest.java

```diff
@@ -0,0 +1,128 @@
+        assertFalse(ALICE_LEAVE.equals(editedAliceLeave));
+
+        // different title -> returns false
+        editedAliceLeave = new LeaveBuilder(ALICE_LEAVE).withTitle("Bob's Paternity Leave").build();
+        assertFalse(ALICE_LEAVE.equals(editedAliceLeave));
+
+        // different description -> returns false
+        editedAliceLeave = new LeaveBuilder(ALICE_LEAVE).withDescription("Bob's Paternity Leave Description").build();
+        assertFalse(ALICE_LEAVE.equals(editedAliceLeave));
+
+        // different start date -> returns false
+        editedAliceLeave = new LeaveBuilder(ALICE_LEAVE).withStart(ALICE_LEAVE.getStart().plusDays(1)).build();
+        assertFalse(ALICE_LEAVE.equals(editedAliceLeave));
+
+        // different end date -> returns false
+        editedAliceLeave = new LeaveBuilder(ALICE_LEAVE).withEnd(ALICE_LEAVE.getEnd().plusDays(1)).build();
+        assertFalse(ALICE_LEAVE.equals(editedAliceLeave));
+    }
+
+    @Test
+    public void isSameLeaveMethod() {
+        // same object -> returns true
+        assertTrue(ALICE_LEAVE.isSameLeave(ALICE_LEAVE));
+
+        // null -> returns false
+        assertFalse(ALICE_LEAVE.isSameLeave(null));
+
+        // different leave -> returns false
+        assertFalse(ALICE_LEAVE.isSameLeave(BOB_LEAVE));
+
+        // different employee -> returns false
+        assertFalse(ALICE_LEAVE.isSameLeave(new LeaveBuilder(ALICE_LEAVE).withEmployee(BOB).build()));
+
+        // different start date -> returns false
+        assertFalse(ALICE_LEAVE.isSameLeave(new LeaveBuilder(ALICE_LEAVE).withStart(ALICE_LEAVE.getStart().plusDays(1))
+                .build()));
+
+        // different end date -> returns false
+        assertFalse(ALICE_LEAVE.isSameLeave(new LeaveBuilder(ALICE_LEAVE).withEnd(ALICE_LEAVE.getEnd().plusDays(1))
+                .build()));
+
+        // same employee, same start date, same end date -> returns true
+        assertTrue(ALICE_LEAVE.isSameLeave(new LeaveBuilder(ALICE_LEAVE).withTitle("Alice's Maternity Leave 2")
+                .withDescription("Alice's Maternity Leave 2 Description").build()));
+
+        // different status -> return true
+        assertTrue(ALICE_LEAVE.isSameLeave(new LeaveBuilder(ALICE_LEAVE).withStatus("Approved").build()));
+    }
+
+    @Test
+    public void toStringMethod() {
+        assertEquals(ALICE_LEAVE.toString(), "Employee: " + ALICE.getName()
+                + " Title: " + ALICE_LEAVE.getTitle()
+                + " Start: " + ALICE_LEAVE.getStart()
+                + " End: " + ALICE_LEAVE.getEnd()
+                + " Status: " + ALICE_LEAVE.getStatus());
+    }
+
+    @Test
+    public void belongsToMethod() {
+        // same employee -> returns true
+        assertTrue(ALICE_LEAVE.belongsTo(ALICE));
+
+        // different employee -> returns false
+        assertFalse(ALICE_LEAVE.belongsTo(BOB));
+    }
+
+    @Test
+    public void hashCodeMethod() {
```

Comment:
If you want, you can try varying each field of ALICE_LEAVE and check that the hashcode is different